### PR TITLE
Add tool to check upgrade tags

### PR DESCRIPTION
**Pull request by liangxia**
The tag mismatch between `upgrade-prepare` and `upgrade-check` cause upgrade CI jobs to fail, and then affect auto-release jobs. So add this tool to check upgrade tags.

Local usage,
```
$ find features/upgrade/ -name "*.feature" -exec go run tools/cmd/check-upgrade-tags/main.go {} \;
```

After this PR merged, I'd like to introduce a new job into OpenShift CI to gate the PR merge in this repo whenever there are changes in folder `features/upgrade`

### CODE REVIEW

**Comment by liangxia** [comment]
Tested the code with PR https://github.com/openshift/verification-tests/pull/3706

**Comment by pruan-rht** [comment]
@liangxia overall it looks ok, but a couple of areas of note:
1. Golang spacing is 8 not 2.  Currently your file has mixed 2 and 4 spacing.  I recommend using and IDE plugin that will autoformat your golang code (VSCode is a very good choice)
2. try to make variable names to be easier to understand...for example, it's hard to tell what that variable `oris` or `oriLines` mean.

**Comment by liangxia** [comment]
> @liangxia overall it looks ok, but a couple of areas of note:
> 
> 1. Golang spacing is 8 not 2.  Currently your file has mixed 2 and 4 spacing.  I recommend using and IDE plugin that will autoformat your golang code (VSCode is a very good choice)
> 2. try to make variable names to be easier to understand...for example, it's hard to tell what that variable `oris` or `oriLines` mean.

That's quite strange.  I'm using VSCode, and it auto formats the codes and does not complains.

**Comment by pruan-rht** [comment]
@liangxia which plugin are you using?  I'm using this one  https://code.visualstudio.com/docs/languages/go .

**Comment by liangxia** [comment]
> @liangxia which plugin are you using? I'm using this one https://code.visualstudio.com/docs/languages/go .

There is a `.editconfig` that set tab width to 2 spaces for `.go`, https://github.com/openshift/verification-tests/blob/3f52e4e012572abd8b0aa8f2a54668c3e8853ec2/.editorconfig#L21

File: tools/cmd/check-upgrade-tags/main.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			}
+		}
+	}
+}
+
+// writeFile writes to a file, by reading from map[int]string
+// for a specific line,
+// if sedmap is not empty, write that line.
+// if addmap is not empty, add that line, and then add the line from orimap.
+func writeFile(filepath string, orimap, addmap, sedmap *map[int]string) {
```
[the remark this answers is not in the document]
Comment (pruan-rht) [reply tools/cmd/check-upgrade-tags/main.go:106]: @liangxia I see some inconsistencies when it comes to your naming conventions.  In Golang, it's recommended to use camelCase., so `addmap` would becomd `addMap` and etc.

[the remark this answers is not in the document]
Comment (liangxia) [reply tools/cmd/check-upgrade-tags/main.go]: Updated

**Pull request merged**